src/usermod.c

```diff
@@ -308,13 +308,13 @@ static struct ulong_range getulong_range(const char *str)
 	errno = 0;
 	first = strtoll(str, &pos, 10);
 	if (('\0' == *str) || ('-' != *pos ) || (ERANGE == errno) ||
-	    (first != (unsigned long int)first))
+	    (first != (long long)(unsigned long int)first))
 		goto out;
 
 	errno = 0;
 	last = strtoll(pos + 1, &pos, 10);
 	if (('\0' != *pos ) || (ERANGE == errno) ||
-	    (last != (unsigned long int)last))
+	    (last != (long long)(unsigned long int)last))
 		goto out;
 
 	if (first > last)
```

Comment:
I can't link to the last lines of this function, but the casts there are useless and should be removed:

```c
	result.first = (unsigned long int)first;
	result.last = (unsigned long int)last;
```

The destination is of that type, so the conversion is automatic.

Reply:
The casts check for truncation on architectures where `long long` and `unsigned long` are of different size.

Reply:
I meant the casts like `result.last = (unsigned long int)last;` (not the ones written by you).  Casts in an assignment will not add warnings.